examples/tuners/dngo_tuner/pybnn/bayesian_linear_regression.py

```diff
@@ -0,0 +1,298 @@
+import emcee
+import logging
+import numpy as np
+
+from scipy import optimize
+from scipy import stats
+
+from pybnn.base_model import BaseModel
+
+
+def linear_basis_func(x):
+    return np.append(x, np.ones([x.shape[0], 1]), axis=1)
+
+
+def quadratic_basis_func(x):
+    x = np.append(x ** 2, x, axis=1)
+    return np.append(x, np.ones([x.shape[0], 1]), axis=1)
+
+
+logger = logging.getLogger(__name__)
+
+
+class Prior(object):
+
+    def __init__(self, rng=None):
+        if rng is None:
+            self.rng = np.random.RandomState(np.random.randint(0, 10000))
+        else:
+            self.rng = rng
+
+    def lnprob(self, theta):
+        """
+        Compute the log probability for theta = [log alpha, log beta]
+        :param theta:
+        :return: log p(theta)
+        """
+        lp = 0
+        lp += stats.norm.pdf(theta[0], loc=0, scale=1)  # log alpha
+        lp += stats.norm.pdf(theta[1], loc=0, scale=1)  # log sigma^2
+
+        return lp
+
+    def sample_from_prior(self, n_samples):
+        p0 = np.zeros([n_samples, 2])
+
+        # Log alpha
+        p0[:, 0] = self.rng.normal(loc=0,
+                                   scale=1,
+                                   size=n_samples)
+
+        # Log sigma^2
+        p0[:, 1] = self.rng.normal(loc=-3,
+                                   scale=1,
+                                   size=n_samples)
+        return p0
+
+
+class BayesianLinearRegression(BaseModel):
```

Comment:
Why can't we use bayesian regression from third-party library (e.g., sklearn) so that we don't have to maintain one on our own?

Reply:
Can we make pybnn our dependency? `https://github.com/automl/pybnn`

Reply:
We do not use the whole https://github.com/automl/pybnn, only the dngo-related part is included. 
So I am not sure about "Can we make pybnn our dependency?"

Reply:
I mean, I can `pip install pybnn` and `import pybnn`, rather than copy the whole `pybnn` into our repo.